### PR TITLE
fix azure keyvault tests

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

azure keyvault tests was failed because
1. added to wrong group
2. try to access invalid tenant configuration

fix those 2 test issues.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

integration tests of azure key vault.

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/yungez/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION

@yuwzho  @zikalino 

```

```

### CODE REVIEW

**Comment by sivel** [comment]
From #38130 this is failing due to:

azure_rm_keyvaultkey and azure_rm_keyvaultsecrets module testing failed, exposed below code bugs:

* no dependency to azure-keyvault
* invalid credential passed to keyvaultClient.

detail error pls see https://app.shippable.com/github/ansible/ansible/runs/59616/62/console

**Comment by yungezz** [comment]
@mattclay, need your help to get azure test service principal's object_id. here's the context:
when assigning  azure key vault access policy to one service principal, you need specify object_id, instead of application_id of the service principal. that's one of reason keyvault test failed to create key, secret etc. To get object id of the test service principal, pls run below az cli command:
```
az ad sp show --id <test_service_principal_application_id>
```

**Comment by mattclay** [comment]
@yungezz The service principal is different for each test run, so the object_id will need to be looked up in the test. A lookup plugin should be created for this so we don't need to install the Azure CLI .

**Comment by yungezz** [comment]
@mattclay  thanks for the info. but there's a hardcoded client_id in existing tests, https://github.com/ansible/ansible/blob/devel/test/integration/targets/azure_rm_keyvaultkey/tasks/main.yml#L17. how it works?

**Comment by yungezz** [comment]
ready_for_review    @zikalino  @yuwzho  @jborean93

**Review by zikalino — APPROVED** [review]
looks good to me

**Review by mattclay — CHANGES_REQUESTED** [review]
@yungezz Please rebase this PR and remove the `disabled` aliases from the integration tests. The existing tests do not pass. CI is not reporting failures because the tests are currently disabled.

**Comment by mattclay** [comment]
needs_rebase

**Comment by yungezz** [comment]
@mattclay  rebased and enabled keyvault key and secret tests. now the test failed because of the issue I mentioned in previous comments: need to grant vault permission to test service principal object_id. So the question is to how to get the object_id. You mentioned that it will need a lookup plugin. Can we do this in 2 PRs? Is it possible to merge this PR firstly with test disabled, then enabled it after lookup plugin merged?

**Comment by yungezz** [comment]
and currently keyvault tests contains a hardcode id https://github.com/ansible/ansible/blob/devel/test/integration/targets/azure_rm_keyvaultkey/tasks/main.yml#L17, can we get object_id so pass the tests, then adding plugin later?

**Comment by nitzmahone** [comment]
Yeah, the lookup plugin thing isn't going to work very well anyway, since you'd have to pass all the auth args to it today. We've been kicking around a "cloud as inventory object" thing that might solve that, but it wouldn't happen until probably 2.8 at the earliest. 

As we've discussed before, this module's effectively useless in the real world in its current incarnation without the ability to look up IDs. So I'm not too worried about the tests being disabled- I don't think anyone's going to be able to use it anyway. Might be better to focus on some of the other modules for now and revisit this one when there's time to properly solve the lookup issues (probably inside the module impl itself for 2.7).

If you really want to get it going, there would need to be a custom module just for the test (eg, something in the library/ dir under the test role) that could look up the necessary object IDs.

**Comment by yungezz** [comment]
@nitzmahone  thanks for comments. I've tested the fixing manually by get object id with az cli. user could do same thing to get object id by running shell in ansible after installing az cli.  But as Matt clay mentioned ansible test don't want to take that way by installing az cli. I'm trying to add a lookup plugin to get azure service principal's object id. You're right, user will need to pass in all azure credentials.  Is there any concern to let user pass in azure credentials? I'm thinking user will lookup those credentials via lookup plugin.

**Comment by yungezz** [comment]
@mattclay  HI Matt, after adding azure service principal object id lookup plugin, keyvault key and secrets test passed. while CI failled at some job trying to install supervisor package https://app.shippable.com/github/ansible/ansible/runs/67626/36/console. no idea how this related to my change. Could you help to have a look? Thanks.

**Comment by yungezz** [comment]
split plugin to a seperate pr as @jborean93  suggested.

**Comment by yungezz** [comment]
@mattclay , @nitzmahone   azure_service_pricipal_attribute lookup plugin is seperated to pr https://github.com/ansible/ansible/pull/40932 per suggestion.

File: test/integration/targets/azure_rm_keyvaultkey/tasks/main.yml [hunk 1]
```diff
@@ -1,20 +1,35 @@
 - name: Prepare random number
   set_fact:
     rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+    tenant_id: "{{ lookup('env','AZURE_TENANT') }}"
```
Comment (yuwzho) [inline test/integration/targets/azure_rm_keyvaultkey/tasks/main.yml:4]: Move this line after line 10?

**Pull request closed without merging**